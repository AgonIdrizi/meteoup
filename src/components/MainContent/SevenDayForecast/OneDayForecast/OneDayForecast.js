import React from 'react';
import PropTypes from 'prop-types';
import classes from './OneDayForecast.module.scss';


const propTypes = {
    
};


const oneDayForecast = (props) => {

    
    
    return (
        <div style={props.style} onClick={e => props.clicked(e, props.id)} className={classes.OneDayForecast}>
           <div className='Date'>
              <div>
                  <span>Mon</span>
              </div>
              <div>
                  <span>6/24</span>
              </div>
           </div>
           <div className="Image">
               <div title="Showers">
                  <img alt="Showers, thunderstorms likely" src="https://www.meteoblue.com/website/images/picto/08_iday.svg" height="50px" />
               </div>
           </div>
           <div className="Degrees">
               <div className="DayTemp"> <span>27</span></div>
               <div className="NightTemp"><span>18</span></div>
           </div>
           <div className="WindRainSunInfo">
               <div className="Wind">
                   <span>..</span>
                   <span>15 km/h</span>
               </div>
               <div className="Rain">
                   <span>...</span>
                   <span>0-10 mm</span>
               </div>
               <div>
                   <span>..</span>
                   <span>3 h</span>
               </div>
           </div>
           <div className="Info-color"></div>
        </div>
    );
};


oneDayForecast.propTypes = propTypes;


export default oneDayForecast;
