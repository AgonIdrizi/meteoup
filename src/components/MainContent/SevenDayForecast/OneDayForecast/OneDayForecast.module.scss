.OneDayForecast {
    position: relative;
    display:flex;
    flex-direction: column;
    width: 100%;
    align-items: center;
    text-align: center;
    border: 1px solid grey;
    
}