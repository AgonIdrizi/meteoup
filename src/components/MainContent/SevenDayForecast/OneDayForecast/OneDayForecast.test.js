import React from 'react'

import { render, cleanup, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import OneDayForecast from './OneDayForecast'


afterEach(cleanup)

const data = {
    "date": "2019-07-27",
    "date_epoch": 1564185600,
    "day": {
        "maxtemp_c": 30.3,
        "maxtemp_f": 86.5,
        "mintemp_c": 12.2,
        "mintemp_f": 54.0,
        "avgtemp_c": 23.7,
        "avgtemp_f": 74.6,
        "maxwind_mph": 4.7,
        "maxwind_kph": 7.6,
        "totalprecip_mm": 3.3,
        "totalprecip_in": 0.13,
        "avgvis_km": 9.6,
        "avgvis_miles": 5.0,
        "avghumidity": 57.0,
        "condition": {
            "text": "Moderate or heavy rain shower",
            "icon": "//cdn.apixu.com/weather/64x64/day/356.png",
            "code": 1243
        },
        "uv": 9.2
    },
    "astro": {
        "sunrise": "05:24 AM",
        "sunset": "08:01 PM",
        "moonrise": "01:01 AM",
        "moonset": "03:31 PM"
    }
}
    
    it('renders correctly', () => {
        const { asFragment, getByTestId } = render(<OneDayForecast data={data} />);
        expect(asFragment()).toMatchSnapshot();
        //expect(getByTestId())
    })

