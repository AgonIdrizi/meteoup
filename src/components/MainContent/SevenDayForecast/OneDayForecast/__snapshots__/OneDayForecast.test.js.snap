// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="OneDayForecast"
  >
    <div
      class="Date"
    >
      <div>
        <span>
          Sat
        </span>
      </div>
      <div>
        <span>
          7/27
        </span>
      </div>
    </div>
    <div
      class="Image"
      style="background-color: rgb(202, 219, 234); width: 100%;"
    >
      <div
        title="Moderate or heavy rain shower"
      >
        <img
          alt="Moderate or heavy rain shower"
          height="50px"
          src="//cdn.apixu.com/weather/64x64/day/356.png"
        />
      </div>
    </div>
    <div
      class="Degrees"
      style="width: 100%;"
    >
      <div
        class="DayTemp"
        style="width: 100%; background-color: rgb(220, 83, 48);"
      >
         
        <span>
          30.3 째C
        </span>
      </div>
      <div
        class="NightTemp"
        style="width: 100%; background-color: rgb(131, 234, 139);"
      >
        <span>
          12.2 째C
        </span>
      </div>
    </div>
    <div
      class="WindRainSunInfo"
    >
      <div
        class="Wind"
      >
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-wind fa-w-16 "
            data-icon="wind"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M156.7 256H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h142.2c15.9 0 30.8 10.9 33.4 26.6 3.3 20-12.1 37.4-31.6 37.4-14.1 0-26.1-9.2-30.4-21.9-2.1-6.3-8.6-10.1-15.2-10.1H81.6c-9.8 0-17.7 8.8-15.9 18.4 8.6 44.1 47.6 77.6 94.2 77.6 57.1 0 102.7-50.1 95.2-108.6C249 291 205.4 256 156.7 256zM16 224h336c59.7 0 106.8-54.8 93.8-116.7-7.6-36.2-36.9-65.5-73.1-73.1-55.4-11.6-105.1 24.9-114.9 75.5-1.9 9.6 6.1 18.3 15.8 18.3h32.8c6.7 0 13.1-3.8 15.2-10.1C325.9 105.2 337.9 96 352 96c19.4 0 34.9 17.4 31.6 37.4-2.6 15.7-17.4 26.6-33.4 26.6H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16zm384 32H243.7c19.3 16.6 33.2 38.8 39.8 64H400c26.5 0 48 21.5 48 48s-21.5 48-48 48c-17.9 0-33.3-9.9-41.6-24.4-2.9-5-8.7-7.6-14.5-7.6h-33.8c-10.9 0-19 10.8-15.3 21.1 17.8 50.6 70.5 84.8 129.4 72.3 41.2-8.7 75.1-41.6 84.7-82.7C526 321.5 470.5 256 400 256z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          7.6 km/h
        </span>
      </div>
      <div
        class="Rain"
      >
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-rain fa-w-16 "
            data-icon="cloud-rain"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: rgb(3, 111, 144);"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.1 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96zM88 374.2c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          3.3 mm
        </span>
      </div>
      <div>
        <span>
          ..
        </span>
        <span>
          3 h
        </span>
      </div>
    </div>
    <div
      class="Info-color"
    />
  </div>
</DocumentFragment>
`;

exports[`test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="OneDayForecast"
  >
    <div
      class="Date"
    >
      <div>
        <span>
          Sat
        </span>
      </div>
      <div>
        <span>
          7/27
        </span>
      </div>
    </div>
    <div
      class="Image"
      style="background-color: rgb(202, 219, 234); width: 100%;"
    >
      <div
        title="Moderate or heavy rain shower"
      >
        <img
          alt="Moderate or heavy rain shower"
          height="50px"
          src="//cdn.apixu.com/weather/64x64/day/356.png"
        />
      </div>
    </div>
    <div
      class="Degrees"
      style="width: 100%;"
    >
      <div
        class="DayTemp"
        style="width: 100%; background-color: rgb(220, 83, 48);"
      >
         
        <span>
          30.3 째C
        </span>
      </div>
      <div
        class="NightTemp"
        style="width: 100%; background-color: rgb(131, 234, 139);"
      >
        <span>
          12.2 째C
        </span>
      </div>
    </div>
    <div
      class="WindRainSunInfo"
    >
      <div
        class="Wind"
      >
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-wind fa-w-16 "
            data-icon="wind"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M156.7 256H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h142.2c15.9 0 30.8 10.9 33.4 26.6 3.3 20-12.1 37.4-31.6 37.4-14.1 0-26.1-9.2-30.4-21.9-2.1-6.3-8.6-10.1-15.2-10.1H81.6c-9.8 0-17.7 8.8-15.9 18.4 8.6 44.1 47.6 77.6 94.2 77.6 57.1 0 102.7-50.1 95.2-108.6C249 291 205.4 256 156.7 256zM16 224h336c59.7 0 106.8-54.8 93.8-116.7-7.6-36.2-36.9-65.5-73.1-73.1-55.4-11.6-105.1 24.9-114.9 75.5-1.9 9.6 6.1 18.3 15.8 18.3h32.8c6.7 0 13.1-3.8 15.2-10.1C325.9 105.2 337.9 96 352 96c19.4 0 34.9 17.4 31.6 37.4-2.6 15.7-17.4 26.6-33.4 26.6H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16zm384 32H243.7c19.3 16.6 33.2 38.8 39.8 64H400c26.5 0 48 21.5 48 48s-21.5 48-48 48c-17.9 0-33.3-9.9-41.6-24.4-2.9-5-8.7-7.6-14.5-7.6h-33.8c-10.9 0-19 10.8-15.3 21.1 17.8 50.6 70.5 84.8 129.4 72.3 41.2-8.7 75.1-41.6 84.7-82.7C526 321.5 470.5 256 400 256z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          7.6 km/h
        </span>
      </div>
      <div
        class="Rain"
      >
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-cloud-rain fa-w-16 "
            data-icon="cloud-rain"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: rgb(3, 111, 144);"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.1 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96zM88 374.2c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0zm160 0c-12.8 44.4-40 56.4-40 87.7 0 27.7 21.5 50.1 48 50.1s48-22.4 48-50.1c0-31.4-27.2-43.1-40-87.7-2.2-8.1-13.5-8.5-16 0z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span>
          3.3 mm
        </span>
      </div>
      <div>
        <span>
          ..
        </span>
        <span>
          3 h
        </span>
      </div>
    </div>
    <div
      class="Info-color"
    />
  </div>
</DocumentFragment>
`;
